store/messages.js

```diff
@@ -12,183 +13,231 @@ const SEND_MESSAGE_ERROR = "SEND_MESSAGE_ERROR";
 
 // ---------- ACTION CREATORS ---------- //
 
-export const getMessages = messages => ({ type: GET_MESSAGES, messages });
-const addMessage = message => ({ type: ADD_MESSAGE, message });
+export const getMessages = (messages) => ({ type: GET_MESSAGES, messages });
+const addMessage = (message) => ({ type: ADD_MESSAGE, message });
 // const sendMessageError = (message) => ({ type: ADD_CONTACT_ERROR, message });
 
 // ---------- THUNK CREATORS ---------- //
 
 // GET MESSAGES FOR CURRENT CHAT
 export const fetchMessages = () => (dispatch, getState) => {
-	// query for all messages for the current chat, and add listener on child_added for new messages
-	if (getState().chats.currentChat) {
-		db.ref(`messages/${getState().chats.currentChat.id}`).on("child_added", function(snapshot) {
-			// format a message object compatible with GiftedChat, message text not added yet
-			const newMessage = {
-				_id: snapshot.key,
-				user: {
-					_id: snapshot.val().senderId,
-					name: snapshot.val().senderName
-				},
-				createdAt: snapshot.val().timestamp,
-				original: snapshot.val().translations.original
-			};
-
-			const userLanguage = getState().user.language;
-
-			// if the message was sent by the user it will not be translated
-			if (snapshot.val().senderId !== getState().firebase.auth.uid) {
-				// check if translation to user's language exists
-				if (snapshot.val().translations[userLanguage]) {
-					newMessage.text = snapshot.val().translations[userLanguage];
-					dispatch(addMessage(newMessage));
-				} else {
-					// translate the original message to the language of the user
-					fetch(
-						`https://translation.googleapis.com/language/translate/v2?q=${snapshot.val().message}&target=${getLangKey(
-							userLanguage
```

Comment:
The key is still in the git history, fyi

Reply:
I'll look into deleting past commits